nav2_dwb_controller/dwb_plugins/src/kinematic_parameters.cpp

```diff
@@ -107,6 +108,11 @@ void KinematicsHandler::initialize(
   nh->get_parameter(plugin_name + ".decel_lim_y", kinematics.decel_lim_y_);
   nh->get_parameter(plugin_name + ".decel_lim_theta", kinematics.decel_lim_theta_);
 
+  kinematics.max_vel_x_default_ = kinematics.max_vel_x_;
```

Comment:
For absolute linear speed limit in DWB, `max_speed_xy_` variable is responsible. `max_vel_x_` and `max_vel_y_` are responsible for maximum OX and OY component of linear velocity. By its logic, the algorithm is storing default values of absolute maximum velocity and components max. velocity before it will modify them by message from `SpeedFilter`. I am doubt that `absolute_*` is applicable for these parameters. `default_` here means default values before modification and which will be restored after robot will leave speed restriction area.

Reply:
OK

Reply:
Made `entire_` prefix in https://github.com/ros-planning/navigation2/commit/3465c90415afd11418f98fc01bbc9386afa853ea